--- charts-original/templates/_helpers.tpl
+++ charts/templates/_helpers.tpl
@@ -38,4 +38,12 @@
 {{- if .Values.podLabels }}
 {{- toYaml .Values.podLabels | nindent 8 }}
 {{- end }}
-{{- end -}}
\ No newline at end of file
+{{- end -}}
+
+{{- define "system_default_registry" -}}
+{{- if .Values.global.cattle.systemDefaultRegistry -}}
+{{- printf "%s/" .Values.global.cattle.systemDefaultRegistry -}}
+{{- else -}}
+{{- "" -}}
+{{- end -}}
+{{- end -}}
