--- charts-original/templates/_helpers.tpl
+++ charts/templates/_helpers.tpl
@@ -215,4 +215,32 @@
   {{- else }}
     {{- include "minio.getValueFromSecret" (dict "Namespace" .Release.Namespace "Name" (include "minio.fullname" .) "Length" 40 "Key" "rootPassword") }}
   {{- end }}
-{{- end -}}
\ No newline at end of file
+{{- end -}}
+
+{{/*
+Windows cluster will add default taint for linux nodes, add below linux tolerations to
+workloads could be scheduled to those linux nodes
+*/}}
+{{- define "linux-node-tolerations" -}}
+- key: "cattle.io/os"
+  value: "linux"
+  effect: "NoSchedule"
+  operator: "Equal"
+{{- end -}}
+
+{{- define "linux-node-selector" -}}
+{{- if semverCompare "<1.14-0" .Capabilities.KubeVersion.GitVersion -}}
+beta.kubernetes.io/os: linux
+{{- else -}}
+kubernetes.io/os: linux
+{{- end -}}
+{{- end -}}
+
+{{/*
+URL prefix for container images to be compatible with Rancher
+*/}}
+{{- define "registry-url" -}}
+{{- if .Values.global.cattle.systemDefaultRegistry -}}
+{{ trimSuffix "/" .Values.global.cattle.systemDefaultRegistry }}/
+{{- end -}}
+{{- end -}}
