--- charts-original/templates/_helpers.tpl
+++ charts/templates/_helpers.tpl
@@ -69,9 +69,32 @@
 {{- if .Values.global.cattle.systemDefaultRegistry -}}
 {{ trimSuffix "/" .Values.global.cattle.systemDefaultRegistry }}/
 {{- else -}}
+{{ if .Values.epinioUI.image.registry }}
 {{ .Values.epinioUI.image.registry }}/
 {{- end -}}
+{{- end -}}
 {{- else -}}
+{{ if .Values.epinioUI.image.registry }}
 {{ .Values.epinioUI.image.registry }}/
 {{- end -}}
 {{- end -}}
+{{- end -}}
+
+{{/*
+Windows cluster will add default taint for linux nodes, add below linux tolerations to
+workloads could be scheduled to those linux nodes
+*/}}
+{{- define "linux-node-tolerations" -}}
+- key: "cattle.io/os"
+  value: "linux"
+  effect: "NoSchedule"
+  operator: "Equal"
+{{- end -}}
+
+{{- define "linux-node-selector" -}}
+{{- if semverCompare "<1.14-0" .Capabilities.KubeVersion.GitVersion -}}
+beta.kubernetes.io/os: linux
+{{- else -}}
+kubernetes.io/os: linux
+{{- end -}}
+{{- end -}}
