--- charts-original/templates/_helpers.tpl
+++ charts/templates/_helpers.tpl
@@ -19,7 +19,7 @@
 Create chart name and version as used by the chart label.
 */}}
 {{- define "kiali-server.chart" -}}
-{{- printf "%s-%s" .Chart.Name .Chart.Version | replace "+" "_" | trunc 63 | trimSuffix "-" }}
+{{- printf "%s" .Chart.Name .Chart.Version | replace "+" "_" | trunc 63 | trimSuffix "-" }}
 {{- end }}
 
 {{/*
@@ -176,3 +176,11 @@
   {{- end }}
 {{- end }}
 {{- end }}
+
+{{- define "system_default_registry" -}}
+{{- if .Values.global.cattle.systemDefaultRegistry -}}
+{{- printf "%s/" .Values.global.cattle.systemDefaultRegistry -}}
+{{- else -}}
+{{- "" -}}
+{{- end -}}
+{{- end -}}
